wren-ai-service/src/providers/engine/wren.py

```diff
@@ -18,30 +18,32 @@ def __init__(self, endpoint: str = os.getenv("WREN_UI_ENDPOINT")):
         self._endpoint = endpoint
         logger.info("Using Engine: wren_ui")
 
-    async def dry_run_sql(
+    async def execute_sql(
         self,
         sql: str,
         session: aiohttp.ClientSession,
         project_id: str | None = None,
+        dry_run: bool = True,
         **kwargs,
     ) -> Tuple[bool, Optional[Dict[str, Any]]]:
+        data = {
+            "sql": remove_limit_statement(sql),
+            "projectId": project_id,
+        }
+        if dry_run:
+            data["dryRun"] = True
+            data["limit"] = 1
+
         async with session.post(
             f"{self._endpoint}/api/graphql",
             json={
                 "query": "mutation PreviewSql($data: PreviewSQLDataInput) { previewSql(data: $data) }",
-                "variables": {
-                    "data": {
-                        "dryRun": True,
-                        "limit": 1,
-                        "sql": remove_limit_statement(sql),
-                        "projectId": project_id,
-                    }
-                },
+                "variables": {"data": data},
             },
         ) as response:
             res = await response.json()
-            if res.get("data"):
-                return True, None
```

Comment:
how about keep the error and return one more value for data? I think using the same position to represent two meaning might be not a good choice.

e.g. 
```py
return status, data, error
```